### PR TITLE
cmake: Default to installing into sysroot

### PR DESCRIPTION
**Pull request by sbc100**
See ChangeLog.md entry for more details.

Also, add a test that uses `cmake --install` to install a library and
the `find_package` to find the library config.

This confirms that #6595 is fixed even though technicall the fix landed
in in #14291

### CODE REVIEW

File: ChangeLog.md
Comment (kripken) [inline ChangeLog.md:25]: I'm not sure what this means. That `make install` will copy to the sysroot? Or something else?

Comment (sbc100) [reply ChangeLog.md:25]: Yes.

Previously the default was to try to install to `/usr/local` on the host system.. which would either give you "Permission Denied" or install into a place you probably never want to store emscripten libraries.

Comment (sbc100) [reply ChangeLog.md:25]: I will try to elaborate.

Comment (sbc100) [reply ChangeLog.md]: I wrote a much longer ChangeLog.md entry.

I think I will wait until we release 2.0.23 before landing since I'd like to get the std libs updates into their own separate release.

File: ChangeLog.md
Comment (kripken) [inline ChangeLog.md:24]:
```suggestion
  directly) are now configured to install (by default) directly into the emscripten
```

File: ChangeLog.md
Comment (kripken) [inline ChangeLog.md:29]:
```suggestion
  install into the host system (e.g /use/local) which is almost always not
```
```suggestion
  install into the host system (e.g /usr/local) which is almost always not
```

File: ChangeLog.md
Comment (kripken) [inline ChangeLog.md:31]:
```suggestion
  their own secondary sysroot may be able to simplify their build system by
```

File: tests/test_other.py
Comment (kripken) [inline tests/test_other.py:689]:
```suggestion
    # Build an application that uses find_package to locate and use the above library.
```

**Review by kripken — APPROVED** [review]
Plan sounds good.

**Pull request merged**
